classes/reviewForm/ReviewFormDAO.inc.php

```diff
@@ -35,12 +35,17 @@ function getById($reviewFormId, $assocType = null, $assocId = null) {
 
 		$result = $this->retrieve (
 			'SELECT	rf.*,
-				SUM(CASE WHEN ra.date_completed IS NOT NULL THEN 1 ELSE 0 END) AS complete_count,
-				SUM(CASE WHEN ra.review_id IS NOT NULL AND ra.date_completed IS NULL THEN 1 ELSE 0 END) AS incomplete_count
-			FROM	review_forms rf
-				LEFT JOIN review_assignments ra ON (ra.review_form_id = rf.review_form_id AND ra.declined<>1)
-			WHERE	rf.review_form_id = ? AND rf.assoc_type = ? AND rf.assoc_id = ?
-			GROUP BY rf.review_form_id',
+				aggregation.complete_count,
+				aggregation.incomplete_count
+			FROM review_forms rf
```

Comment:
I suspect this is working around SQL Server issues with `SUM(CASE(...))`, correct? This query has given us trouble in earlier formulations around performance. Unfortunately I don't have an especially large test dataset to work with. How much content do you have there?

Reply:
It is mostly to prevent the keywords "group by" having a lot of parameters. But I read somewhere that doing the aggregation in the join was better overall. We don't really have a huge dataset for now. The performance looks good at this point.